src/sst/core/simulation.cc

```diff
@@ -596,6 +596,14 @@ void Simulation_impl::setup() {
 
 }
 
+#ifdef EVENT_PROFILING
+std::map<std::string, uint64_t> eventHandlers;
+std::map<std::string, uint64_t> eventRecvCounters;
+std::map<std::string, uint64_t> eventSendCounters;
+uint64_t messageSizeSent = 0;
+uint64_t messageSizeRecv = 0;
```

Comment:
Can't have these as globals.  I think this will work today because simulation construction is serialized across threads.  However, we will probably need to move to parallel construction for really large simulations at some point, at which time these would be accessed concurrently by multiple threads.  You're going to have to move these data structures into the Simulation objects themselves so that they are limited to access by a single thread at a time.

Actually, on further inspection, it looks like there is the possibility these may be accessed concurrently by different threads.